### PR TITLE
The big updoot: One Year Later

### PR DESCRIPTION
**Pull request by Michiyamenotehifunana**
## About The Pull Request
It's about time we started catching up on our backlog of PRs again. Better late than never.

This PR will port over the cover rework, RIG ricochets, combat diving (with many fixes), and throwing knives. Also you can shoot through stairs now.

## Why It's Good For The Game

This will modernize our combat just a bit more and definitely breathe new life into the server.

## Changelog
```changelog
add: Diving. Hold C in hotkey mode while moving to dive in any direction. You can dive on tables, railings, and low walls. Diving into glass hurts you, but it also hurts the glass. (You'll get hurt more than the glass does)
add: RIG ricochets. Your RIGs can now deflect bullets.
add: Throwing knives. Craft them.
add: Cover rework, low walls, tables, and security barricades now block bullets somewhat more immersively.
add: Bullets now travel up and down stairs.
add: In order to accommodate diving, you can now hold items while laying down. You are unable to attack or use items while laying down, however. There is also now a delay for getting up from laying down.
```

### CODE REVIEW

**Comment by Jamini** [comment]
Tori, we voted against RIG ricochetes.

**Comment by Michiyamenotehifunana** [comment]
riggochets disabled also im blaming jams for the weird dme conflict

**Comment by Michiyamenotehifunana** [comment]
Rebased.

File: code/modules/mob/inventory/slots.dm [hunk 1]
```diff
@@ -97,10 +97,10 @@
...
 
 /datum/inventory_slot/hand/can_equip(obj/item/I, mob/living/carbon/human/owner, disable_warning)
-	if(owner.lying && I.canremove)
-		if(!disable_warning)
-			to_chat(owner, SPAN_WARNING("You can't hold items while lying"))
-		return FALSE
+//	if(owner.lying && I.canremove)
+//		if(!disable_warning)
+//			to_chat(owner, SPAN_WARNING("You can't hold items while lying"))
+//		return FALSE
```
Comment (Shadow-Quill) [inline code/modules/mob/inventory/slots.dm:103]: Mention this change in the changelog please.

File: code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm [hunk 2]
```diff
@@ -18,31 +18,31 @@
 	reagents.maximum_volume = 40
 
 
-/*/mob/living/carbon/superior_animal/roach/benzin/Life()
+/mob/living/carbon/superior_animal/roach/benzin/Life()
 	if(reagents.total_volume < reagents.maximum_volume)
 		reagents.add_reagent("fuel", 1)
-	..()*/
+	..()
```
Comment (Shadow-Quill) [inline code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm:24]: This needs to be re-commented out. It'd be appreciated if an `OCCULUS EDIT` comment could be tacked on as well.

File: code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm [hunk 3]
```diff
@@ -18,31 +18,31 @@
...
 
 
 /mob/living/carbon/superior_animal/roach/benzin/bullet_act(obj/item/projectile/P, def_zone)
 	. = ..()
-	if(prob(40))
+	if(prob(80))
```
Comment (Shadow-Quill) [inline code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm:29]: This needs to be mentioned in the PR.

File: code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm [hunk 4]
```diff
@@ -18,31 +18,31 @@
...
 
-/*/mob/living/carbon/superior_animal/roach/benzin/attackby(obj/item/I, mob/living/user, params)
+
+/mob/living/carbon/superior_animal/roach/benzin/attackby(obj/item/I, mob/living/user, params)
 	if(user.a_intent == I_HELP && istool(I))
 		var/obj/item/tool/T = I
 		if(T.use_fuel_cost)
 			return FALSE
 	. = ..()
-*/
```
Comment (Shadow-Quill) [inline code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm:38]: Comment out.

File: code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm [hunk 5]
```diff
@@ -18,31 +18,31 @@
...
 /mob/living/carbon/superior_animal/roach/benzin/fire_act()
 	if(stat != DEAD)
 		explosion(get_turf(src), -1, -1, 2, 3)
 
-/*
+
 /mob/living/carbon/superior_animal/roach/benzin/death()
 	. = ..()
 	new /obj/effect/decal/cleanable/liquid_fuel(loc, reagents.get_reagent_amount("fuel"), 1)
-*/
```
Comment (Shadow-Quill) [inline code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm:48]: Comment out.

**Review by Shadow-Quill — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm (hunk 2 above)
Comment (Michiyamenotehifunana) [reply code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm]: Regression fixed

File: code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm (hunk 4 above)
Comment (Michiyamenotehifunana) [reply code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm]: Regression fixed

File: code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm (hunk 5 above)
Comment (Michiyamenotehifunana) [reply code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm]: Regression fixed

File: code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm (hunk 3 above)
Comment (Michiyamenotehifunana) [reply code/modules/mob/living/carbon/superior_animal/roach/types/benzin.dm]: Accidental regression, fixed now.

File: code/modules/mob/inventory/slots.dm (hunk 1 above)
Comment (Michiyamenotehifunana) [reply code/modules/mob/inventory/slots.dm:103]: Now mentioned.

**Review by Shadow-Quill — APPROVED** [review]

**Pull request merged**